pkg/controller/common.go

```diff
@@ -158,60 +166,66 @@ func (c *SharedState) getPodsFromVolume(volumeName string) ([]string, error) {
 	return pods, nil
 }
 
-func (c *SharedState) getNodesForReplica(ctx context.Context, azr azReconciler, volumes []string, pods ...string) ([]string, error) {
-	if volumes == nil {
-		volumes = []string{}
-		for _, pod := range pods {
-			podVolumes, err := c.getVolumesFromPod(pod)
-			if err != nil {
-				return nil, err
-			}
-			volumes = append(volumes, podVolumes...)
-		}
-	}
-
-	numReplica := 0
-	for _, volume := range volumes {
-		var azVolume v1alpha1.AzVolume
-		if err := azr.getClient().Get(ctx, types.NamespacedName{Namespace: azr.getNamespace(), Name: volume}, &azVolume); err != nil {
+func (c *SharedState) getVolumesForPods(pods ...string) ([]string, error) {
+	volumes := []string{}
+	for _, pod := range pods {
+		podVolumes, err := c.getVolumesFromPod(pod)
+		if err != nil {
 			return nil, err
 		}
-		numReplica = max(numReplica, azVolume.Spec.MaxMountReplicaCount)
+		volumes = append(volumes, podVolumes...)
 	}
+	return volumes, nil
+}
 
+func getRankedNodesForReplicaAttachments(ctx context.Context, azr azReconciler, volumes []string) ([]string, error) {
+	klog.V(5).Info("Getting ranked list of nodes for creating AzVolumeAttachments")
+	numReplica := 0
 	nodes := []string{}
 	nodeScores := map[string]int{}
 	primaryNode := ""
+
 	for _, volume := range volumes {
-		volReq, err := createLabelRequirements(azureutils.VolumeNameLabel, volume)
+		azVolume, err := azureutils.GetAzVolume(ctx, azr.getClient(), azr.getAzClient(), volume, consts.AzureDiskCrdNamespace, true)
 		if err != nil {
+			klog.V(5).Info(fmt.Sprintf("AzVolume for volume %s is not found.", volume))
```

Comment:
You could use [`Infof`](https://pkg.go.dev/k8s.io/klog#Verbose.Infof) instead of `Info(fmt.Sprintf(...))`.